errbot/backends/text.py

```diff
@@ -113,6 +105,24 @@ def __init__(self, config):
         self.user = self.build_identifier(self.bot_config.BOT_ADMINS[0])
 
     def serve_forever(self):
+        try:
+            # Load readline for better editing/history behaviour
+            import readline
+
+            # Implement a simple completer for commands
+            def completer(txt, state):
+                options = [i for i in self.all_commands if i.startswith(txt)] if txt else list(self.all_commands.keys())
+                if state < len(options):
+                    return options[state]
+
+            readline.parse_and_bind("tab: complete")
+            readline.set_completer(completer)
+
+        except ImportError:
+            # Readline is Unix-only
+            log.debug("Python readline module is not available")
+            pass
+
```

Comment:
Can we move this block into its own function so it doesn't pollute `serve_forever()` so much?